src/crimson/osd/osd.cc

```diff
@@ -158,7 +158,7 @@ CompatSet get_osd_initial_compat_set()
 
 seastar::future<> OSD::open_meta_coll()
 {
-  return store.open_collection(
+  return store.get_local_shard_store().open_collection(
```

Comment:
Worth a TODO:
// TODO: encapsulate into OSDMeta to use its FuturizedShardStore

Or fix it directly in this PR if convenient.